test/unit/advertisement-operator/broadcaster_test.go

```diff
@@ -344,10 +345,16 @@ func TestNotifyAdvertisementDeletion(t *testing.T) {
 	}
 	// create adv on foreign cluster
 	adv := prepareAdv(&b)
+	adv.Status.AdvertisementStatus = advtypes.AdvertisementAccepted
+	adv.Finalizers = append(adv.Finalizers, advop.FinalizerString)
 	adv2, _ := b.SendAdvertisementToForeignCluster(adv)
+
 	// modify adv status to DELETING
 	err = b.NotifyAdvertisementDeletion()
-	time.Sleep(1 * time.Second)
 	assert.Nil(t, err)
-	assert.Equal(t, advtypes.AdvertisementDeleting, adv2.Status.AdvertisementStatus)
+
```

Comment:
My 2 cents: Wait for an event on a certain resource instead of sleeping would be a better practice.